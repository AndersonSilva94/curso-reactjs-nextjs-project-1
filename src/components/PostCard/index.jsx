import React, { Component } from 'react';
import './style.css'


class PostCard extends Component {
  render() {
    const { post = {} } = this.props;
    const  { title, body, cover} = post;
    return (
      <div className="post">
        <img src={cover} alt={title} />
        <div className="post-card">
          <h2> {title} </h2>
          <p> {body} </p>
        </div>
    </div>
    );
  }
}

export default PostCard;

/* export const PostCard = ({ post }) => {
  // console.log(props)
  const { title, body, cover } = post;
  return (
    
  );
};
 */