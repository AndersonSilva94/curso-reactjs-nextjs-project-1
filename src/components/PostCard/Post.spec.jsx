import { render, screen } from '@testing-library/react';
import { postCardsPropsMock } from './mock';
import PostCard from './index';

const props = postCardsPropsMock

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard {...props} />);

    /* expect(screen.getByRole('img', { name: props.title})).toBeInTheDocument(); */
    expect(screen.getByAltText(/title 1/i)).toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', { name: 'title 1' })).toBeInTheDocument();
    expect(screen.getByText('body 1')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  })
})