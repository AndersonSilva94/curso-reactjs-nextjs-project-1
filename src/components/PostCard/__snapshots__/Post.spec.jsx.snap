// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img />
  <div
    class="post-card"
  >
    <h2>
       
       
    </h2>
    <p>
       
       
    </p>
  </div>
</div>
`;
